trajoptlib/include/trajopt/constraint/LaneConstraint.hpp

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) TrajoptLib contributors
+
+#pragma once
+
+#include "trajopt/constraint/PointLineRegionConstraint.hpp"
+#include "trajopt/util/SymbolExports.hpp"
+
+namespace trajopt {
+
+/**
+ * Lane Constraint.
+ *
+ * Specifies the robot must stay between two lines.
+ */
+class TRAJOPT_DLLEXPORT LaneConstraint {
+ public:
+  /**
+   * Constructs a LaneConstraint.
+   *
+   * @param topLine The top constraint line. Side must be equal to top.
+   * @param bottomLine The bottom constraint line. Side must be equal to bottom.
```

Comment:
You could enforce the side invariants by taking four points and constructing the PointLineRegionConstraints inside this constructor instead.